src/Microsoft.AspNetCore.Razor.Design/Internal/AssemblyTagHelperDescriptorResolver.cs

```diff
@@ -118,5 +126,103 @@ private void EnsureMvcInitialized()
             _populateMethodDelegate = (Action<IServiceCollection, string>) populateMethod
                 ?.CreateDelegate(typeof(Action<IServiceCollection, string>));
         }
+
+        private IEnumerable<TagHelperDescriptor> GetFakeDescriptors(IEnumerable<TagHelperDescriptor> vcthDescriptors)
+        {
+            var fakeDescriptors = new List<TagHelperDescriptor>();
+            foreach (var descriptor in vcthDescriptors)
+            {
+                var fakeType = typeof(ViewComponentTagHelperDesignTimeType).FullName;
+                var fakeDescriptor = new TagHelperDescriptor(descriptor);
+                fakeDescriptor.TypeName = fakeType;
+
+                var fakeAttributes = new List<TagHelperAttributeDescriptor>();
+                fakeDescriptor.Attributes = fakeAttributes;
+
+                var fakeHelperType = typeof(ViewComponentTagHelperDesignTimeHelperType).FullName;
+                var fakeHelperDescriptor = new TagHelperDescriptor(descriptor);
+                fakeHelperDescriptor.TypeName = fakeHelperType;
+
+                var fakeHelperAttributes = new List<TagHelperAttributeDescriptor>();
+                fakeHelperDescriptor.Attributes = fakeHelperAttributes;
+
+                foreach (var attribute in descriptor.Attributes)
+                {
+                    var summary = $"{attribute.TypeName}: {descriptor.TypeName}.{attribute.PropertyName}";
+                    var fakeVariableName = $"__obj";
+                    var fakeAttributePropertyName = $"{nameof(ViewComponentTagHelperDesignTimeType.PlaceholderProperty)} = null; " +
+                            $"{fakeType}.{nameof(ViewComponentTagHelperDesignTimeType.ActionProperty)} = () => {{ " +
+                            $"{attribute.TypeName} {fakeVariableName} = default({attribute.TypeName}); " +
+                            $"{fakeType}.{nameof(ViewComponentTagHelperDesignTimeType.PlaceholderMethod)}({fakeVariableName}); {fakeVariableName} ";
+
+                    var fakeAttribute = new TagHelperAttributeDescriptor()
```

Comment:
Side issue: We should add copy ctors to `TagHelperAttributeDescriptor` and `TagHelperRequiredAttributeDescriptor`, design time descs etc.